Formula/git-archive-all.rb

```diff
@@ -19,6 +19,9 @@ def install
     rewrite_shebang detected_python_shebang, "*.py"
 
     system "make", "prefix=#{prefix}", "install"
+
+    inreplace bin/"git-archive-all", "#! /usr/bin/env python",
+                                     "#!#{Formula["python@3.10"].opt_bin/"python3"}"
```

Comment:
looks like `rewrite_shebang` wasn't doing anything here 
```suggestion
    rewrite_shebang detected_python_shebang, "git_archive_all.py"

    system "make", "prefix=#{prefix}", "install"
```

Reply:
Agreed, that should solve the problem. Do you think it's worth modifying the `rewrite_shebang` method to support globbing like this? Looks like this is the only formula that has this problem ([`swift`](https://github.com/Homebrew/homebrew-core/blob/f96e45468c688cf53bd99786be53d265e1316abd/Formula/swift.rb#L362) gets around it by expanding the wildcard path before passing to `rewrite_shebang`) so I'm thinking let's not bother.

Reply:
Yup I agree that it wouldn't be worth effort.

One thing we might consider though is to throw an exception if no replacements occurred (which would almost certainly be a bug, like it was here)

Reply:
Agreed!

Reply:
Awesome, I can make a pr. It'll be my first one to `brew` 😄, would you mind reviewing when it's ready?

Reply:
🎉 Awesome, sure! I might not have time until tomorrow though...

Reply:
I got confused and thought maybe the `git_archive_all` was not a .py file in the source archive, or something wrong was going on. Of course this was still using Python 2 on macOS and we would have noticed once Apple removes Python 2.

I updated the code accordingly.

I also agree that having an error when there is no rewrite would be great :) Thanks all.